api/src/main/java/org/openmrs/config/DatabaseConfig.java

```diff
@@ -0,0 +1,109 @@
+package org.openmrs.config;
+
+import org.springframework.context.annotation.*;
+import org.springframework.core.env.Environment;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.JpaVendorAdapter;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+
+import javax.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+import java.util.Properties;
+
+@Configuration
+@EnableJpaRepositories("org.openmrs.api.db")
+@PropertySource("classpath:hibernate.default.properties")
+public class DatabaseConfig {
+	
+	@Bean
+	LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource, Environment env) {
```

Comment:
My understanding is that this will result in us having both entityManagerFactory and hibernateSessionFactory, thus using up extra resources and adding complexity. Please see http://stackoverflow.com/a/21673855 for the alternative.